packages/artillery/lib/platform/cloud/cloud.js

```diff
@@ -127,6 +137,18 @@ class ArtilleryCloudPlugin {
     global.artillery.ext({
       ext: 'onShutdown',
       method: async (opts) => {
+        const waitTimeInMs = 11000; //at most wait more than 10 seconds (report interval is 10s)
```

Comment:
Would on `onShutdown` ever get called before all VUs are finished and reports are received? How likely is it that Artillery is now going to take 10s+ to stop every time?

Reply:
Actually @hassy , now that I think about it, do we still need that `await sleep(5 * 1000);` there, at least to such a large number? Was it put there to avoid this precise problem? Perhaps with this implementation it can also be reduced, or is no longer needed?

Reply:
Actually @hassy I've been thinking about this a little bit more - since you implemented it, do you remember if there's a reason other than this exact one why that `await sleep(5 * 1000)` is there? 

If not, then I think perhaps we might be able to remove it (or greatly reduce it), as this new mechanism now waits for the actual events to be sent. So if there wasn't another reason, it might be an opportunity to revisit it?